src/uu/vmstat/src/vmstat.rs

```diff
@@ -4,187 +4,129 @@
 // file that was distributed with this source code.
 
 mod parser;
+mod picker;
 
-use clap::{crate_version, Command};
+#[cfg(target_os = "linux")]
+use crate::picker::{get_pickers, Picker};
+#[allow(unused_imports)]
+use clap::{arg, crate_version, ArgMatches, Command};
 #[allow(unused_imports)]
 pub use parser::*;
-#[cfg(target_os = "linux")]
-use procfs::{Current, CurrentSI};
-use uucore::error::UResult;
+#[allow(unused_imports)]
+use uucore::error::{UResult, USimpleError};
 use uucore::{format_usage, help_about, help_usage};
 
 const ABOUT: &str = help_about!("vmstat.md");
 const USAGE: &str = help_usage!("vmstat.md");
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let _matches = uu_app().try_get_matches_from(args)?;
-
-    let mut section: Vec<String> = vec![];
-    let mut title: Vec<String> = vec![];
-    let mut data: Vec<String> = vec![];
-    let mut data_len_excess = 0;
-
+    #[allow(unused)]
+    let matches = uu_app().try_get_matches_from(args)?;
     #[cfg(target_os = "linux")]
-    let func = [
-        concat_helper(get_process_info),
-        concat_helper(get_memory_info),
-        concat_helper(get_swap_info),
-        concat_helper(get_io_info),
-        concat_helper(get_system_info),
-        concat_helper(get_cpu_info),
-    ];
-
-    #[cfg(not(target_os = "linux"))]
-    let func: [ConcatFunc; 0] = [];
-
-    func.iter()
-        .for_each(|f| f(&mut section, &mut title, &mut data, &mut data_len_excess));
-
-    println!("{}", section.join(" "));
-    println!("{}", title.join(" "));
-    println!("{}", data.join(" "));
+    {
+        // validate unit
+        if let Some(unit) = matches.get_one::<String>("unit") {
+            if !["k", "K", "m", "M"].contains(&unit.as_str()) {
+                Err(USimpleError::new(
+                    1,
+                    "-S requires k, K, m or M (default is KiB)",
+                ))?;
+            }
+        }
+
+        let one_header = matches.get_flag("one-header");
+        let no_first = matches.get_flag("no-first");
+        let delay = matches.get_one::<String>("delay");
+        let mut delay = if let Some(delay) = delay {
+            let delay = delay.parse::<i64>().unwrap();
+            if delay <= 0 {
+                return Err(USimpleError::new(-1, "delay must be greater than 0"));
+            }
+            delay
+        } else {
+            -1
+        };
+        let count = matches.get_one::<String>("count");
+        let mut count = if let Some(count) = count {
+            let mut count = count.parse::<i64>().unwrap();
+            if count < 0 {
+                return Err(USimpleError::new(-1, "count must be greater than 0"));
+            }
+            if count == 0 {
+                count = 1;
+            }
+            count
+        } else {
+            -1
+        };
```

Comment:
The handling of both `delay` and `count` will panic if you provide an invalid value like "invalid". And as the values can't be negative, I would use `u64` instead of `i64`. By using a value parser, you can let `clap` handle the parsing.